From 2f21b3a1b99f6ecfba54c1e50833556cdf8e180b Mon Sep 17 00:00:00 2001
From: Ulf Hansson <ulf.hansson@linaro.org>
Date: Tue, 14 Jan 2014 17:36:21 +0100
Subject: [PATCH 062/202] mmc: core: Use generic CMD6 time while switching to
 eMMC HS200 mode

upstream commit 57de31f6351ecad9212a26d2112b546e4c8b4a5
Author: Ulf Hansson <ulf.hansson@linaro.org>

Conform to the eMMC spec and use the CMD6 generic timeout from the
EXT_CSD register, when switching to HS200 mode.

Signed-off-by: Ulf Hansson <ulf.hansson@linaro.org>
Signed-off-by: Chris Ball <chris@printf.net>
Signed-off-by: Peter Swain <pswain@cavium.net>
[Original patch taken from Cavium SDK 3.1.1-544]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 drivers/mmc/core/mmc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/core/mmc.c b/drivers/mmc/core/mmc.c
index 3821e2a..7f0eab5 100644
--- a/drivers/mmc/core/mmc.c
+++ b/drivers/mmc/core/mmc.c
@@ -856,7 +856,8 @@ static int mmc_select_hs200(struct mmc_card *card)
 	/* switch to HS200 mode if bus width set successfully */
 	if (!err)
 		err = mmc_switch(card, EXT_CSD_CMD_SET_NORMAL,
-				 EXT_CSD_HS_TIMING, 2, 0);
+				EXT_CSD_HS_TIMING, 2,
+				card->ext_csd.generic_cmd6_time);
 err:
 	return err;
 }
-- 
1.8.2.1

